bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/internal/base/restrictions/BaseRestriction.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.base.restrictions;
+
+import java.util.Objects;
+
+import org.eclipse.passage.lic.internal.api.LicensedProduct;
+import org.eclipse.passage.lic.internal.api.diagnostic.TroubleCode;
+import org.eclipse.passage.lic.internal.api.requirements.Requirement;
+import org.eclipse.passage.lic.internal.api.restrictions.Restriction;
+
+@SuppressWarnings("restriction")
+public final class BaseRestriction implements Restriction {
+
+	private final LicensedProduct product;
+	private final Requirement requirement;
+	private final TroubleCode reason;
+
+	public BaseRestriction(LicensedProduct product, Requirement requirement, TroubleCode reason) {
+		Objects.requireNonNull(product);
```

Comment:
Will not be clear who is `null`, let's add a message with the field name as usual.